Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -6746,12 +6756,62 @@ bool IWindowExplorer.TestElementLocators(ElementInfo EI, bool GetOutAfterFoundEl
                 }
             }
             finally
+            {
+                Driver.Manage().Timeouts().ImplicitWait = (TimeSpan.FromSeconds((int)ImplicitWait));
+            }
+        }
+
+
+        void IWindowExplorer.CollectOriginalElementsDataForDeltaCheck(ObservableList<ElementInfo> mOriginalList)
+        {
+            try
+            {
+                mIsDriverBusy = true;
+                Driver.Manage().Timeouts().ImplicitWait = new TimeSpan(0, 0, 0);
+
+                foreach (ElementInfo EI in mOriginalList)
+                {
+                    EI.ElementStatus = ElementInfo.eElementStatus.Pending;
+                }
+
+
+                foreach (ElementInfo EI in mOriginalList)
+                {
+                    try
+                    {
+                        SwitchFrame(EI);
+                        IWebElement e = LocateElementByLocators(EI.Locators);
+                        if (e != null)
+                        {
+                            EI.ElementObject = e;
+                            EI.ElementStatus = ElementInfo.eElementStatus.Passed;
+                        }
+                        else
+                        {
+                            EI.DeltaStatus = ElementInfo.eDeltaStatus.Deleted;
+                            EI.ElementStatus = ElementInfo.eElementStatus.Failed;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        EI.DeltaStatus = ElementInfo.eDeltaStatus.Deleted;
+                    }
+                }
+            }
+            finally
             {
                 Driver.Manage().Timeouts().ImplicitWait = (TimeSpan.FromSeconds((int)ImplicitWait));
                 Driver.SwitchTo().DefaultContent();
                 mIsDriverBusy = false;
             }
+        }
+
+        public ElementInfo GetMatchingElement(ElementInfo latestElement, ObservableList<ElementInfo> originalElements)
+        {
+
```

Comment:
why compare been done on all latestElement.ElementObject, we said you will compare only the e.ID
comparing objects is not good practice